### PR TITLE
IA-1707 / IA-1708: convert MarkerMap & GeoJsonMap class component to functional components

### PR DESCRIPTION
**Pull request by codeKameleon**
Replace MarkerMap and GeoJsonMap By functional components

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes
- replace markerMap
- delete Polygon map by GeoJsonMap
- library to parse [WKT](https://fr.wikipedia.org/wiki/Well-known_text) as a GeoJson and then support multi polygon. Previously the map was crashing with multi polygons
- split `logCompareComponent`
- Create `Value` component

## Print screen / video

[209983173-26a252af-397d-4b35-a4c5-5a9214b37a65.webm](https://user-images.githubusercontent.com/12494624/210250086-fd464238-e2a1-4215-9809-dfe147412e68.webm)

Screenshot 2023-01-02 at 11 37 55

### CODE REVIEW

**Comment by beygorghor** [comment]
@codeKameleon i did some more deep changes. It was not working with multi polygon at all or Point using [WKT](https://fr.wikipedia.org/wiki/Well-known_text)

**Comment by beygorghor** [comment]
@cypress

File: hat/assets/js/apps/Iaso/components/logs/Value.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const styles = theme => ({
+    cellMap: {
+        margin: -theme.spacing(2),
+    },
+});
+
+const useStyles = makeStyles(styles);
+
+export const Value: FunctionComponent<Props> = ({fieldKey, value}) => {
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/logs/Value.tsx]: Not the best name ever...

File: hat/assets/js/apps/Iaso/components/logs/Value.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const classes = useStyles();
+    if (!value || value.toString().length === 0) return textPlaceholder;
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/logs/Value.tsx]: I don't understand the string conversion. Based on the typing, value can be:
- a string
- a number, but then I don't see any stringified number with a length of 0
- undefined: in which case `value.toString()` will throw an error

File: hat/assets/js/apps/Iaso/components/logs/ValueWithErrorBoundary.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            );
+        }
+        return <Value fieldKey={fieldKey} value={value} />;
+    }
+}
+ValueWithErrorBoundary.defaultProps = {
+    value: undefined,
+};
+ValueWithErrorBoundary.propTypes = {
+    value: PropTypes.any,
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/logs/ValueWithErrorBoundary.js]: propType is something like `PropTypes.oneOfType(PropTypes.string, PropTypes.number)`

File: hat/assets/js/apps/Iaso/domains/orgUnits/utils.js [hunk 4]
```diff
@@ -9,22 +9,6 @@ import {
...
-        .split(',');
-    const polygonPositions = [];
-    positionsArrays.forEach(pos => {
-        const lat = pos.split(' ')[0];
-        const lng = pos.split(' ')[1];
-        polygonPositions.push([lng, lat]);
-    });
-    return polygonPositions;
-};
-
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/orgUnits/utils.js:27]: Is this what the new terraformer dep does?

**Comment by olethanh** [comment]
are the cypress fixed now? I think it was in main?

**Comment by beygorghor** [comment]
@cypress

File: hat/assets/js/apps/Iaso/domains/orgUnits/utils.js (hunk 4 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/orgUnits/utils.js:27]: yes, this code is from trypelim ....

File: hat/assets/js/apps/Iaso/components/logs/ValueWithErrorBoundary.js (hunk 3 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/components/logs/ValueWithErrorBoundary.js:59]: it can be an object or an array or a boolean also...

**Pull request merged**